src/ring_theory/power_series/basic.lean

```diff
@@ -1010,6 +1011,35 @@ begin
   { intros m hm, rwa nat.eq_zero_of_le_zero (nat.le_of_succ_le_succ hm) }
 end
 
+open finset nat
+
+/-- The ring homomorphism taking a power series `f(X)` to `f(aX)`. -/
+noncomputable def eval_mul_hom (a : R) : power_series R →+* power_series R :=
+{ to_fun :=   λ f, power_series.mk $ λ n, a^n * (power_series.coeff R n f),
+  map_zero' := by { ext, simp only [linear_map.map_zero, power_series.coeff_mk, mul_zero], },
+  map_one' := by { ext1, simp only [mul_boole, power_series.coeff_mk, power_series.coeff_one],
+                split_ifs, { rw [h, pow_zero], }, refl, },
+  map_add' := by {intros, ext, exact mul_add _ _ _, },
+  map_mul' := by {intros, ext, rw [power_series.coeff_mul, power_series.coeff_mk,
+              power_series.coeff_mul, finset.mul_sum], apply sum_congr rfl,
+              simp only [coeff_mk, prod.forall, nat.mem_antidiagonal],
+              intros b c H, rw [<-H, pow_add, mul_mul_mul_comm] }, }
```

Comment:
style: again, the `apply` is easier to spot if you don't tack it on the same line as the wrapped `rw` list`.
```suggestion
  map_mul' := λ f g, by {
    ext,
    rw [power_series.coeff_mul, power_series.coeff_mk, power_series.coeff_mul, finset.mul_sum],
    apply sum_congr rfl,
    simp only [coeff_mk, prod.forall, nat.mem_antidiagonal],
    intros b c H,
    rw [<-H, pow_add, mul_mul_mul_comm] }, }
```